It implements `esrally-storage prune` sub-command.

fressi-elastic opened the pull request:
This following up PR #1959

The main purpose is cleaning up all local files written by storage manager on the local directories. It also allows to list file names using the `ls` subcommand.

Example of use:

`esrally-storage ls --filenames` It lists all downloaded data filenames.
`esrally-storage ls --status-filenames` It lists all status filenames.
`esrally-storage ls --filenames --status-filenames` It lists all filenames.

`esrally-storage prune` It deletes all downloaded and status files.
`esrally-storage ls --filenames` It deletes all downloaded data files.
`esrally-storage ls --status-filenames` It deletes all status files.

fressi-elastic committed:
WIP: Add per-mirror statistics to transfer manager.

fressi-elastic committed:
WIP

fressi-elastic committed:
Add Google Cloud Storage adapter.

fressi-elastic committed:
Add unit test for GSAdapter class.

fressi-elastic committed:
Update requirements.

fressi-elastic committed:
Update uv lock file.

fressi-elastic committed:
GSAdapter use the same thread pool executor as transfer manager.

fressi-elastic committed:
Mock default credentials.

fressi-elastic committed:
Fix support for Python <= 3.12

fressi-elastic committed:
Add code documentation.

fressi-elastic committed:
Rename dependency group: gcs->gs

fressi-elastic committed:
It wraps  object so that we do have one single legacy implementation for  which doesn't relies on  method.

fressi-elastic committed:
Read blob chunks from the same thread that iterates chunks.

fressi-elastic committed:
Remove unrelated changes.

fressi-elastic committed:
Remove stuff imported from _executor from the main package module.

fressi-elastic committed:
Make testing monkey patching more robust to changes in the internal Google client classes.

fressi-elastic committed:
Remove value for DEFAULT_GOOGLE_CLOUD_PROJECT

fressi-elastic committed:
Merge branch 'master' of github.com:elastic/rally into storage

fressi-elastic committed:
Pass PyLint check.

fressi-elastic committed:
Pass unit tests.

fressi-elastic committed:
Merge branch 'storage.gc' into storage

fressi-elastic committed:
Fix CLI and integration test.

fressi-elastic committed:
Remove status property from Tranfer class.

fressi-elastic committed:
Refactor import section in unit tests.

fressi-elastic committed:
Merge branch 'master' of github.com:elastic/rally into storage.gc

fressi-elastic committed:
Rebase GSAdapter on top of HTTPAdapter

fressi-elastic committed:
Rebase GSAdapter on top of HTTPAdapter

fressi-elastic committed:
Use AnonymousCredentials when default Google APIs credentials are not available.

fressi-elastic committed:
Merge branch 'storage.gc' into storage

fressi-elastic committed:
Move range constrained file writer to its own module.

fressi-elastic committed:
Add unit test for FileWriter class.

fressi-elastic committed:
Add functional test to verify files download.

fressi-elastic committed:
Add integration test to verify files download.

fressi-elastic committed:
Use anycrc library to implement CRC32C `Checksum.combine` method.

fressi-elastic committed:
Verify finished document once transfer is complete.

fressi-elastic committed:
Fix Transfer unit test

fressi-elastic committed:
It adds s3:// anf gs:// cases to test_adapter_registry_get.

fressi-elastic committed:
Merge branch 'storage.gc' into storage

fressi-elastic committed:
Merge branch 'master' of github.com:elastic/rally into storage

fressi-elastic committed:
Revert changes to the it package

gbanasiak committed:
Differentiate IT shared setup

fressi-elastic committed:
Fix hang when final transfer file verification fails.

gbanasiak committed:
Document slow marker

elasticmachine committed:
Merge branch 'master' into storage

fressi-elastic committed:
It fixes transfer task finalization.

fressi-elastic committed:
Merge branch 'master' of github.com:elastic/rally into storage

fressi-elastic committed:
Merge branch 'storage' of github.com:fressi-elastic/rally into storage

fressi-elastic committed:
It implements `esrally-storage prune` sub-command.

fressi-elastic committed:
It adds IT test case for `prune` sub-command.

fressi-elastic committed:
Merge branch 'master' of github.com:elastic/rally into storage.prune

gbanasiak commented on esrally/storage/_cli.py:
```diff
@@ -47,16 +47,19 @@ def main():
+    ls_parser = subparsers.add_parser("ls", help="It lists file(s) downloaded from ES Rally remote storage services.")
+    get_parser = subparsers.add_parser("get", help="It downloads file(s) from ES Rally rem1ote storage services.")
+    put_parser = subparsers.add_parser("put", help="It uploads file(s) to mirror server.")
+    prune_parser = subparsers.add_parser("prune", help="It delete transfer files from local directories.")
```
```suggestion
    prune_parser = subparsers.add_parser("prune", help="It deletes transfer files from local directories.")
```

fressi-elastic commented on esrally/storage/_transfer.py:
```diff
@@ -727,3 +727,17 @@ def _check_finished_document(self) -> None:
+    def prune(self):
+        self.close()
+        errors: list[Exception] = []
+        for p in [self.path, self.status_file_path]:
```
OK no problem. It is easy to add a flag like `--status-files` to select only those files.

fressi-elastic committed:
It foollows ups PR #1959 code review.

- It updates CLI help messages.
- It gets monitor interval from configuration and/or command line options.
- It fixes other things in the storage client and test cases.

fressi-elastic committed:
Allow to list/delete transfer file types by CLI distinguishing between status files and data files.

fressi-elastic committed:
Update IT test cases for CLI module.

gbanasiak reviewed (approved):
LGTM
